File: ontrack-model/src/test/java/net/nemerosa/ontrack/model/support/tree/support/MarkupTest.java
package net.nemerosa.ontrack.model.support.tree.support;

import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;

public class MarkupTest {

    @Test
    public void text() {
        Markup m = Markup.text("Test");
        assertNull(m.getType());
        assertEquals("Test", m.getText());
        assertNull(m.getAttributes());
    }

    @Test(expected = NullPointerException.class)
    public void text_no_attr() {
        Markup m = Markup.text("Test");
        m.attr("test", "xxx");
    }

}

File: ontrack-model/src/test/java/net/nemerosa/ontrack/model/support/tree/support/MarkupTest.kt
package net.nemerosa.ontrack.model.support.tree.support

import net.nemerosa.ontrack.model.support.tree.support.Markup.Companion.text
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals
import kotlin.test.assertNull

class MarkupTest {
    @Test
    fun text() {
        val m = text("Test")
        assertNull(m.type)
        assertEquals("Test", m.text)
        assertNull(m.attributes)
    }
}
